File: lib/erl/src-loose/transport/tTransportException.erl
-module(tTransportException).

-include("thrift/thrift.hrl").
-include("thrift/transport/tTransportException.hrl").

-export([new/0, new/1, new/2, message/1]).

new(Type, Message) ->
    #tTransportException{type = Type, message = Message}.

new()     -> new(?tTransportException_UNKNOWN, nil). % WATCH
new(Type) -> new(Type, nil). % WATCH

message(This) ->
    ?ATTR(message).

File: lib/csharp/src/Transport/TTransportException.cs
//
//  TTransportException.cs
//
//  Begin:  Aug 19, 2007
//  Authors: 
//		Todd Berman <tberman@imeem.com>
//
//  Copyright (C) 2007 imeem, inc. <http://www.imeem.com>
//  All rights reserved.
//

using System;
using System.Collections.Generic;
using System.Text;

namespace Thrift.Transport
{
	public class TTransportException : Exception
	{
		protected ExceptionType type;

		public TTransportException()
			: base()
		{
		}

		public TTransportException(ExceptionType type)
			: this()
		{
			this.type = type;
		}

		public TTransportException(ExceptionType type, string message)
			: base(message)
		{
			this.type = type;
		}

		public TTransportException(string message)
			: base(message)
		{
		}

		public ExceptionType Type
		{
			get { return type; }
		}

		public enum ExceptionType
		{
			Unknown,
			NotOpen,
			AlreadyOpen,
			TimedOut,
			EndOfFile
		}
	}
}
